File: frontend/src/app/shared/models/note-access-right.ts
/**
 * Запись о предоставлении пользователю прав доступа к заметке
 */
export interface NoteAccessRight {
  /**
   * Id заметки
   */
  noteId: string;
  /**
   * Id пользователя, которому предоставляются права
   */
  userId: string;
  /**
   * Права доступа
   */
  rights: 'read' | 'write' | 'author';
}

File: backend/ISISNotesBackend.Core/Models/NoteAccessRight.cs
using ISISNotesBackend.Core.Models.Enums;

namespace ISISNotesBackend.Core.Models
{
    public class NoteAccessRight
    {
        public string NoteId { get; set; }
        public string UserId { get; set; }
        public string Rights { get; set; }

        public NoteAccessRight(string noteId, string userId, string rights)
        {
            NoteId = noteId;
            UserId = userId;
            Rights = rights;
        }
    }
}